src/sidebar/util/oauth-client.js

```diff
@@ -213,11 +184,44 @@ export default class OAuthClient {
     const headers = {
       'Content-Type': 'application/x-www-form-urlencoded',
     };
-    return fetch(url, {
+    const response = await fetch(url, {
       method: 'POST',
       headers,
       body: params.toString(),
     });
+
+    if (response.status !== 200) {
+      throw new Error('Request failed');
+    }
+
+    return response.json();
+  }
+
+  /**
+   * Fetch an OAuth access token.
+   *
+   * @param {Record<string, string>} data - Parameters for form POST request
```

Comment:
Adding the return may not be that necessary. I don't know. 
```suggestion
   * @param {Record<string, string>} data - Parameters for form POST request
    * @returns {Promise<{accessToken: string, expiresAt: number, refreshToken: string}>}
```

Reply:
I added an explicit `@return` here using the existing `TokenInfo` type.